### PR TITLE
custom email modal to send a mail

### PR DESCRIPTION
**Pull request by nikhilrayaprolu**
in reference to #2874 . I had troubles during squash due to which I made this new branch

### CODE REVIEW

File: app/helpers/helpers.py [hunk 1]
```diff
@@ -109,17 +109,27 @@ def send_session_accept_reject(email, session_name, acceptance, link):
...
-        send_email(
-            to=email,
-            action=SESSION_ACCEPT_REJECT,
-            subject=MAILS[SESSION_ACCEPT_REJECT]['subject'].format(session_name=session_name, acceptance=acceptance),
-            html=MAILS[SESSION_ACCEPT_REJECT]['message'].format(
-                email=str(email),
-                session_name=str(session_name),
-                acceptance=str(acceptance),
-                link=link
+        if request.form:
```
Comment (rafalkowalski) [inline app/helpers/helpers.py]:
```
if request.form:
    subject = request.form.get('subject', '')
    message = request.form.get('message', '')
    action=SESSION_ACCEPT_REJECT
else:
    action=SESSION_ACCEPT_REJECT,
    subject=MAILS[SESSION_ACCEPT_REJECT]['subject'].format(session_name=session_name, acceptance=acceptance),
    message=MAILS[SESSION_ACCEPT_REJECT]['message'].format(
        email=str(email),
        session_name=str(session_name),
        acceptance=str(acceptance),
        link=link)

send_email(
        to=email,
        action=action,
        subject=subject,
        html=message
    )
``` 
This way is much easier

File: app/templates/gentelella/admin/event/sessions/base_session_table.html [hunk 2]
```diff
@@ -196,7 +226,25 @@
...
+            var button = $(event.relatedTarget); // Button that triggered the modal
+            var sessionid = button.data('sessionid'); // Extract info from data-* attributes
+            // If necessary, you could initiate an AJAX request here (and then do the updating in a callback).
+            // Update the modal's content. We'll use jQuery here, but you could use a data binding library or other methods instead.
+            var emailtype = button.data('typeofmail');
+            var sessiontitle = button.data('sessiontitle');
+            var modal = $(this);
+            if (emailtype === "accept") {
+                modal.find('.subject').val("Session " + sessiontitle + " has been accepted");
+                modal.find('.message').val("Hello,The session " + sessiontitle + " has been " + emailtype + "ed by the organizer.  Visit this link to view the session: http://localhost:8001/events/4/sessions/1/");
```
Comment (rafalkowalski) [inline app/templates/gentelella/admin/event/sessions/base_session_table.html]: why is the link hardcoded?

File: app/templates/test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    // Update the modal's content. We'll use jQuery here, but you could use a data binding library or other methods instead.
+    var emailtype = button.data('typeofmail');
+    var sessiontitle = button.data('sessiontitle');
+    var modal = $(this);
+    if (emailtype === "accept") {
+        modal.find('.subject').val("Session " + sessiontitle + " has been accepted");
+        modal.find('.message').val("Hello,The session " + sessiontitle + " has been " + emailtype + "ed by the organizer.  Visit this link to view the session: http://localhost:8001/events/4/sessions/1/");
+        modal.find('.form-horizontal').attr('action', sessionid + '/accept');
+    } else {
+        modal.find('.subject').val("Session " + sessiontitle + " has been rejected");
```
Comment (rafalkowalski) [inline app/templates/test.js]: don't repeat yourself. Please create new function 

```
modal.find('.subject').val("Session " + sessiontitle + " has been accepted");
modal.find('.message').val("Hello,The session " + sessiontitle + " has been " + emailtype + "ed by the organizer.  Visit this link to view the session: http://localhost:8001/events/4/sessions/1/");
modal.find('.form-horizontal').attr('action', sessionid + '/accept');

```

File: app/views/users/sessions.py [hunk 4]
```diff
@@ -141,7 +141,7 @@ def add_speaker_view(event_id, session_id):
         return redirect(url_for('.index_view', event_id=event_id))
 
 
-@event_sessions.route('/<int:session_id>/accept/', methods=('GET',))
+@event_sessions.route('/<int:session_id>/accept/', methods=('GET', 'POST',))
```
Comment (rafalkowalski) [inline app/views/users/sessions.py]: do we need GET method?

File: app/views/users/sessions.py [hunk 5]
```diff
@@ -154,7 +154,7 @@ def accept_session(event_id, session_id):
     return redirect(url_for('.index_view', event_id=event_id))
 
 
-@event_sessions.route('/<int:session_id>/reject/', methods=('GET',))
+@event_sessions.route('/<int:session_id>/reject/', methods=('POST', 'GET',))
```
Comment (rafalkowalski) [inline app/views/users/sessions.py]: do we need GET method?

File: app/views/users/sessions.py [hunk 6]
```diff
@@ -167,7 +167,7 @@ def reject_session(event_id, session_id):
     return redirect(url_for('.index_view', event_id=event_id))
 
 
-@event_sessions.route('/<int:session_id>/send-emails/', methods=('GET',))
+@event_sessions.route('/<int:session_id>/send-emails/', methods=('GET', 'POST'))
```
Comment (rafalkowalski) [inline app/views/users/sessions.py]: do we need GET method?

Comment (rafalkowalski) [inline app/views/users/sessions.py]: do we need GET method?

File: app/views/users/sessions.py (hunk 5 above)
Comment (rafalkowalski) [inline app/views/users/sessions.py]: do we need GET method?

File: app/views/users/sessions.py (hunk 4 above)
Comment (rafalkowalski) [inline app/views/users/sessions.py]: do we need GET method?

**Comment by SaptakS** [comment]
@nikhilrayaprolu the travis build fails. please check.

**Comment by nikhilrayaprolu** [comment]
travis is failing due to change of get request to post.I have modified the tests but  what are the steps to follow to run tests in openevent ? I tried nosetests tests/unittests -v --with-coverage after running the server in background , but this is raising errors.

**Comment by SaptakS** [comment]
@nikhilrayaprolu please ensure that you have nosetests installed. If not install nosetests first. The command to do unittest is `nosetests tests/unittests` only. So no mistake in that.

**Comment by nikhilrayaprolu** [comment]
screenshot from 2017-01-18 12-13-59

@saptakS I tried running  `nosetests tests/unittests` but ended up with the above errors failing all tests

**Comment by niranjan94** [comment]
@nikhilrayaprolu Any progress on this ? This PR has been open for the past 15 days :sweat_smile:

**Comment by niranjan94** [comment]
@nikhilrayaprolu please respond ?

**Comment by nikhilrayaprolu** [comment]
@niranjan94 sorry for the delay, I am allocating time for this now. I will update this thread once I am done.

**Pull request merged**